src/engine/EventManager.cxx

```diff
@@ -304,24 +341,35 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
           {
             m_nodesToPlace.push_back(mouseIsoCoords);
           }
-          bool placementAllowed = false;
+          placementAllowed = false;
+          std::vector<Point> nodesToAdd;
 
           // if we touch a bigger than 1x1 tile also add all nodes of the building to highlight.
           for (const auto &coords : m_nodesToHighlight)
           {
-            Layer layer = TileManager::instance().getTileLayer(tileToPlace);
+            const Layer layer = TileManager::instance().getTileLayer(tileToPlace);
             Point currentOriginPoint = engine.map->getNodeOrigCornerPoint(coords, layer);
+
+            if (currentOriginPoint == UNDEFINED_POINT)
+            { // This should never happen, but check we'll check it anyway just to be safe
+              continue;
+            }
+
```

Comment:
according to my last fixes regarding the original point, this should never happen
because we initialize this point (OriginPoint) after we setNode of a tile, so technically this will never happen.
we can remove it.

Reply:
Removed it